crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs

```diff
@@ -166,18 +171,17 @@ impl OptChaining {
 
         self.vars_with_init = init;
         self.vars_without_init = uninit;
-
-        new
+        *stmts = new;
     }
 }
 
 impl OptChaining {
-    /// Only called from [Fold].
-    fn handle_unary(&mut self, e: UnaryExpr) -> Expr {
+    /// Only called from [VisitMut].
+    fn handle_unary(&mut self, e: &mut UnaryExpr) -> Expr {
         let span = e.span;
 
         if let op!("delete") = e.op {
-            match *e.arg {
+            match &mut *e.arg.take() {
```

Comment:
This is wrong if `e.arg` si not optional chaining expr nor member expr.

Reply:
Thanks, take seems not needed even. 🤦